src/waveform/renderers/glslwaveformrenderersignal.cpp

```diff
@@ -294,9 +304,10 @@ void GLSLWaveformRendererSignal::draw(QPainter* painter, QPaintEvent* /*event*/)
     getGains(&allGain, &lowGain, &midGain, &highGain);
 
     const auto firstVisualIndex = static_cast<GLfloat>(
-            m_waveformRenderer->getFirstDisplayedPosition() * dataSize / 2.0);
+            m_waveformRenderer->getFirstDisplayedPosition() * trackSamples /
+            audioVisualRatio / 2.0);
     const auto lastVisualIndex = static_cast<GLfloat>(
-            m_waveformRenderer->getLastDisplayedPosition() * dataSize / 2.0);
+            m_waveformRenderer->getLastDisplayedPosition() * trackSamples / audioVisualRatio / 2.0);
 
     // const int firstIndex = int(firstVisualIndex+0.5);
     // firstVisualIndex = firstIndex - firstIndex%2;
```

Comment:
There is a check above, that the integer datasize can't be <=1. Therefore `dataSize / 2.0` can't be smaller than 1.0. I wonder if a similar check for `trackSamples / audioVisualRatio / 2.0` is needed?

Reply:
trackSamples and audioVisualRatio are also checked individual.